src/v8n.js

```diff
@@ -37,6 +39,33 @@ function proxyContext(context) {
   });
 }
 
+function proxyContextEs5(context) {
+  Object.keys(availableRules).forEach(prop => {
+    context[prop] = (...args) => {
+      const newContext = proxyContextEs5(context._clone());
+      return newContext._applyRule(availableRules[prop], prop)(...args);
+    };
+  });
+
+  Object.keys(customRules).forEach(prop => {
```

Comment:
This is the exact same as above and a direct repetition. Should be rewritten.

Only change is the set of rules this is running on. I think this could be a loop or a function.